pkg/compare/testdata/OnlyMatchPartial/reference/namespace.yaml

```diff
@@ -0,0 +1,9 @@
+# cluster-compare: allow-undefined-extras=true
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openshift-storage
+  annotations:
+{{ onlyMatch (or .metadata.annotations (dict)) "workload.openshift.io/allowed: management" | indent 4}}
```

Comment:
Just something to keep in mind i think the usage of these functions is harder to read then using the go tempting syntax implementation, i would include the examples on how to do it with template syntax only .
just something to think about it, I will try it out myself and update you :)  @nocturnalastro (not sure about this and need to check it out, i will update you, just wanted you to know that i am concerned from the usage)

Reply:
The main issue I found is that when the template is rendered with the empty map I was getting a issues so had to set a default.

> harder to read then using the go tempting syntax implementation

Do you enumerating through as a loop and setting them to null?

Reply:
was moved to different PR